go-minitwit/db_methods.go

```diff
@@ -230,76 +218,129 @@ func getUserByUsername(userName string) ([]map[string]interface{}, error) {
 
 // registers a new user
 func registerUser(userName string, email string, password [16]byte) error {
-	query := `insert into user (username, email, pw_hash) values (?, ?, ?)`
-	var db, err = connect_db(DATABASE)
-	if err != nil {
-		return err
+
+	pwHashString := hex.EncodeToString(password[:])
+
+	newUser := User{
+		Username: userName,
+		Email:    email,
+		PwHash:   pwHashString,
 	}
-	args := []interface{}{userName, email, pq.Array(password)}
-	messages, err := query_db(db, query, args, false)
-	fmt.Println("this is the messages", messages)
-	return err
+
+	dbNew.Create(&newUser)
+
+	if dbNew.Error != nil {
+		log.Fatal(dbNew.Error)
+		return dbNew.Error
+	}
+
+	return nil
+
 }
 
 // adds a new message to the database
 func addMessage(text string, author_id string) error {
-	query := `insert into message (author_id, text, pub_date, flagged) values (?, ?, ?, 0)`
-	var db, err = connect_db(DATABASE)
+
+	authorIDintValue, err := strconv.Atoi(author_id)
 	if err != nil {
-		fmt.Println("error in addMessage query")
-		return err
+		fmt.Println("Error:", err)
+		return nil
 	}
+
 	currentTime := time.Now().UTC()
 	unixTimestamp := currentTime.Unix()
 
-	args := []interface{}{author_id, text, unixTimestamp, 0}
-	query_db(db, query, args, false)
-	return err
+	newMessage := Message{
+		AuthorID: authorIDintValue,
+		Text:     text,
+		PubDate:  int(unixTimestamp),
+		Flagged:  0, // Default to false for flagged
+	}
+
+	dbNew.Create(&newMessage)
+
+	if dbNew.Error != nil {
+		log.Fatal(dbNew.Error)
+		return dbNew.Error
+	}
+
+	return nil
+
 }
 
 // followUser adds a new follower to the database
 func followUser(userID string, profileUserID string) error {
-	query := `insert into follower (who_id, whom_id) values (?, ?)`
-	var db, err = connect_db(DATABASE)
-	if err != nil {
-		return err
+	userIDInt, errz := strconv.Atoi(userID)
```

Comment:
Same thing, maybe we can change it to int.

Reply:
changing,..